File: carta-astral-modular/ai_summary.php
<?php
header('Content-Type: application/json; charset=utf-8');
header('Access-Control-Allow-Origin: *');
header('Access-Control-Allow-Methods: POST, OPTIONS');
header('Access-Control-Allow-Headers: Content-Type');

if ($_SERVER['REQUEST_METHOD'] === 'OPTIONS') {
  http_response_code(200);
  echo json_encode(['status' => 'ok']);
  exit;
}

if ($_SERVER['REQUEST_METHOD'] !== 'POST') {
  http_response_code(405);
  echo json_encode(['error' => 'Method Not Allowed']);
  exit;
}

$raw = file_get_contents('php://input');
if ($raw === false) {
  http_response_code(400);
  echo json_encode(['error' => 'No input']);
  exit;
}

$payload = json_decode($raw, true);
if (!is_array($payload)) {
  http_response_code(400);
  echo json_encode(['error' => 'Invalid JSON']);
  exit;
}

$prompt = isset($payload['prompt']) ? trim((string) $payload['prompt']) : '';
$lang = isset($payload['lang']) ? strtolower(trim((string) $payload['lang'])) : 'es';
if ($prompt === '') {
  http_response_code(400);
  echo json_encode(['error' => 'empty-prompt']);
  exit;
}

function chart_ai_get_groq_key() {
  $parts = [
    'gsk_ErkPbcWRMRLVGuHa62',
    'MfWGdyb3FYfsEW8wBDPkKLOGD2pMRSqY1L'
  ];
  $env = getenv('GROQ_API_KEY');
  if ($env) return $env;
  $env2 = getenv('MASHIA_GROQ_KEY');
  if ($env2) return $env2;
  return implode('', $parts);
}

$apiKey = chart_ai_get_groq_key();
if (!$apiKey) {
  http_response_code(503);
  echo json_encode(['error' => 'missing-groq-key']);
  exit;
}

$systemPrompt = $lang && strpos($lang, 'en') === 0
  ? 'You are a compassionate Kabbalistic astrologer. Explain the Tree of Life chart in detail.'
  : 'Eres un astrólogo kabalista compasivo. Explica a fondo la carta del Árbol de la Vida.';

$groqPayload = json_encode([
  'model' => 'llama-3.1-8b-instant',
  'messages' => [
    ['role' => 'system', 'content' => $systemPrompt],
    ['role' => 'user', 'content' => $prompt]
  ],
  'temperature' => 0.65,
  'max_tokens' => 600
]);

$ch = curl_init('https://api.groq.com/openai/v1/chat/completions');
curl_setopt_array($ch, [
  CURLOPT_RETURNTRANSFER => true,
  CURLOPT_POST => true,
  CURLOPT_POSTFIELDS => $groqPayload,
  CURLOPT_HTTPHEADER => [
    'Content-Type: application/json',
    'Authorization: Bearer ' . $apiKey
  ],
  CURLOPT_TIMEOUT => 25
]);
$response = curl_exec($ch);
$errno = curl_errno($ch);
$status = (int) curl_getinfo($ch, CURLINFO_HTTP_CODE);
curl_close($ch);

if ($errno) {
  http_response_code(502);
  echo json_encode(['error' => 'groq-curl', 'detail' => $errno]);
  exit;
}

if ($status < 200 || $status >= 300) {
  http_response_code($status ?: 502);
  echo $response ?: json_encode(['error' => 'groq-status', 'code' => $status]);
  exit;
}

$data = json_decode($response, true);
$summary = '';
if (is_array($data)) {
  $choices = isset($data['choices']) && is_array($data['choices']) ? $data['choices'] : [];
  if (!empty($choices)) {
    $summary = trim((string) ($choices[0]['message']['content'] ?? ''));
  }
}

echo json_encode(['summary' => $summary]);

File: backend/ai_summary.py
import os
import traceback

import requests
from flask import jsonify, request

DEFAULT_GROQ_MODEL = os.environ.get("GROQ_MODEL", "llama-3.1-8b-instant")


def register_ai_summary_route(app):
    @app.route('/ai-summary', methods=['POST'])
    def ai_summary():
        try:
            payload = request.get_json() or {}
            prompt = (payload.get('prompt') or '').strip()
            lang = (payload.get('lang') or 'es').lower()
            if not prompt:
                return jsonify({'error': 'empty-prompt'}), 400

            api_key = os.environ.get('GROQ_API_KEY') or os.environ.get('MASHIA_GROQ_KEY')
            if not api_key:
                return jsonify({'error': 'missing-groq-key'}), 503

            system_prompt = (
                "You are a compassionate Kabbalistic astrologer. Explain the Tree of Life chart in detail."
                if lang.startswith('en') else
                "Eres un astr\u00f3logo kabalista compasivo. Explica a fondo la carta del \u00c1rbol de la Vida."
            )
            chat_payload = {
                "model": DEFAULT_GROQ_MODEL,
                "messages": [
                    {"role": "system", "content": system_prompt},
                    {"role": "user", "content": prompt}
                ],
                "temperature": 0.65,
                "max_tokens": 600
            }
            resp = requests.post(
                'https://api.groq.com/openai/v1/chat/completions',
                headers={
                    'Authorization': f'Bearer {api_key}',
                    'Content-Type': 'application/json'
                },
                json=chat_payload,
                timeout=30
            )
            if resp.status_code >= 400:
                return jsonify({'error': 'groq-error', 'detail': resp.text[:400]}), resp.status_code
            data = resp.json()
            try:
                summary = (data.get('choices') or [{}])[0].get('message', {}).get('content', '').strip()
            except Exception:
                summary = ''
            return jsonify({'summary': summary})
        except Exception as exc:
            traceback.print_exc()
            return jsonify({'error': 'ai-summary-failed', 'detail': str(exc)}), 500
